### ISSUE TITLE
Add compatibility range for extensions

### ISSUE THREAD

**Issue by Alkarex** [FreshRSS/FreshRSS#5903]
I don't think there is a way to mark an extension as only compatible with version <= x => ?

_Originally posted by @nicofrand in https://github.com/FreshRSS/FreshRSS/issues/5537#issuecomment-1640256145_

**Comment by aledeg** [comment]
I am wondering if we really need that in FreshRSS. I think it's the extension job to know its compatibility. For instance, we could do something similar to that in the extension:

```php
    public function init(): void {
        if (version_compare(FRESHRSS_VERSION, 1.27) <= 0) {
            $this->registerHook('entry_before_display', [$this, 'deprecatedMethod']);
        } else {
            $this->registerHook('entry_before_display', [$this, 'newMethod']);
        }
    }
```

If we need more validation we could add more branches in the `if`.

**Comment by Alkarex** [comment]
Yes, but we do not have a mechanism at the moment to show in the FreshRSS UI whether a given extension is supposed to work or not (as declared by the extension itself).
